tests/integration/test_application.py

```diff
@@ -51,21 +54,40 @@
 """
 
 
-def test_global_environment(new_dir, monkeypatch, mocker, reset_callbacks):
+def test_global_environment(
+    new_dir,
+    monkeypatch,
+    mocker,
+):
     """Test our additions to the global environment that is available to the
     build process."""
 
-    # Mock some OCI operations that are not relevant to this test.
     rootfs = Path(new_dir) / "rootfs"
     rootfs.mkdir()
     fake_digest = b"deadbeef"
-    mocker.patch.object(oci.Image, "extract_to", return_value=rootfs)
-    mocker.patch.object(oci.Image, "digest", return_value=fake_digest)
+
+    image_info = ImageInfo(
+        base_image=mocker.MagicMock(), base_layer_dir=rootfs, base_digest=fake_digest
+    )
+    mocker.patch.object(RockcraftImageService, "obtain_image", return_value=image_info)
 
     Path("rockcraft.yaml").write_text(ROCKCRAFT_YAML)
 
-    # args = argparse.Namespace(destructive_mode=True)
-    # lifecycle.run("stage", args)
+    monkeypatch.setattr(sys, "argv", ["rockcraft", "prime"])
+    # TODO: we need to do this while we don't support destructive mode
```

Comment:
well....

Reply:
hah I updated this to use `--destructive-mode` and it passed
the system works!